src/App.tsx

```diff
@@ -14,11 +15,85 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+export const App: React.FC = () => {
+  const [isVisible, setIsVisible] = useState(false);
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortBy, setSortBy] = useState('');
 
-export default App;
+  const goodsCopy = [...goodsFromServer];
+
+  goodsCopy.sort((a, b) => {
+    switch (sortBy) {
+      case 'name':
+        return a.localeCompare(b);
+      case 'length':
+        return a.length - b.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    goodsCopy.reverse();
+  }
```

Comment:
it is better to create functions for array sorting and reversing and call functions in onClick